tests/test_ai_router.py

```
@@ -0,0 +1,465 @@
+        with patch.dict(os.environ, {"OLLAMA_ENDPOINT": "http://test.com"}):
+            with patch("urllib.parse.urlparse", side_effect=Exception("Parse error")):
+                with pytest.raises(ValueError, match="Invalid OLLAMA_ENDPOINT URL"):
+                    AIRouter()
+
+
+class TestAdditionalCoverage:
+    """Test additional coverage for uncovered lines"""
+
+    @pytest.mark.asyncio
+    async def test_route_request_forced_ollama_provider(
+        self, ai_router: AIRouter, mock_env_vars: Any
```

Ruff (0.14.1)

Unused method argument: `mock_env_vars`

(ARG002)